src/generateTasks.js

```diff
@@ -14,6 +14,12 @@ module.exports = function generateTasks(config, relFiles) {
   const normalizedConfig = getConfig(config) // Ensure we have a normalized config
   const linters = normalizedConfig.linters
   const globOptions = normalizedConfig.globOptions
+  const ignoreFilters = normalizedConfig.ignore.map(pattern => minimatch.filter(pattern))
```

Comment:
This `minimatch.filter()` call isn't being passed `globOptions`. Is this intentional?

I'm also unable to ignore `'.vscode/*.json'` (whereas `'package.json'` in the same `ignores` array works fine).

Reply:
We should probably pass `globOptions` I guess 🤔 

If we did, then your problem with `.vscode` would probably be resolved because you could specify the `dot` option.

Reply:
I am not able to reproduce this issue locally:
```
λ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        new file:   .vscode/settings.json

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   .lintstagedrc

λ cat .lintstagedrc
{
    "linters": {
        "*.js": [
            "eslint --fix",
            "git add"
        ],
        ".*rc": "jsonlint",
        "*.json": "jsonlint"
    },
    "ignore": [".vscode/*.json"]
}

λ node . --debug
  lint-staged:bin Running `lint-staged@0.0.0-development` +0ms
  lint-staged Loading config using `cosmiconfig` +0ms
  lint-staged Successfully loaded config from `E:\Projects\repos\lint-staged\.lintstagedrc`:
  lint-staged { linters:
  lint-staged    { '*.js': [ 'eslint --fix', 'git add' ],
  lint-staged      '.*rc': 'jsonlint',
  lint-staged      '*.json': 'jsonlint' },
  lint-staged   ignore: [ '.vscode/*.json' ] } +16ms
  lint-staged:cfg Normalizing config +0ms
  lint-staged:cfg Validating config +0ms
Running lint-staged with the following config:
{
  linters: {
    '*.js': [
      'eslint --fix',
      'git add'
    ],
    '.*rc': 'jsonlint',
    '*.json': 'jsonlint'
  },
  ignore: [
    '.vscode/*.json'
  ],
  concurrent: true,
  chunkSize: 9007199254740991,
  globOptions: {
    matchBase: true,
    dot: true
  },
  subTaskConcurrency: 1,
  renderer: 'verbose'
}
  lint-staged Loaded scripts from package.json:
  lint-staged { precommit: 'node index.js',
  lint-staged   cz: 'git-cz',
  lint-staged   lint: 'eslint .',
  lint-staged   'lint:fix': 'yarn lint --fix',
  lint-staged   pretest: 'yarn lint',
  lint-staged   test: 'jest --coverage',
  lint-staged   'test:watch': 'jest --watch' } +16ms
  lint-staged:run Running all linter scripts +0ms
  lint-staged:run Resolved git directory to be `E:\Projects\repos\lint-staged` +15ms
  lint-staged:run Loaded list of staged files in git:
  lint-staged:run [ '.vscode/settings.json' ] +241ms
  lint-staged:gen-tasks Generating linter tasks +0ms
  lint-staged:cfg Normalizing config +272ms
  lint-staged:gen-tasks Generated task:
  lint-staged:gen-tasks { pattern: '*.js',
  lint-staged:gen-tasks   commands: [ 'eslint --fix', 'git add' ],
  lint-staged:gen-tasks   fileList: [] } +0ms
  lint-staged:gen-tasks Generated task:
  lint-staged:gen-tasks { pattern: '.*rc', commands: 'jsonlint', fileList: [] } +0ms
  lint-staged:gen-tasks Generated task:
  lint-staged:gen-tasks { pattern: '*.json', commands: 'jsonlint', fileList: [] } +0ms
Running tasks for *.js [started]
Running tasks for .*rc [started]
Running tasks for *.json [started]
Running tasks for *.js [skipped]
→ No staged files match *.js
Running tasks for .*rc [skipped]
→ No staged files match .*rc
Running tasks for *.json [skipped]
→ No staged files match *.json
  lint-staged linters were executed successfully! +271ms
```

Please take a look at your debug logs. It could be a configuration issue. If you continue to face the issue, please do create a new issue.